pennylane/pulse/parametrized_evolution.py

```diff
@@ -316,6 +438,16 @@ def fun(y, t):
             """dy/dt = -i H(t) y"""
             return (-1j * H_jax(self.data, t=t)) @ y
 
-        result = odeint(fun, y0, self.t, **self.odeint_kwargs)
-        mat = result[-1]
+        mat = odeint(fun, y0, self.t, **self.odeint_kwargs)
+        if self.return_intermediate:
+            # Skip U(t_1, t_1) for both activated and deactivated `complementary`
+            mat = mat[1:]
```

Comment:
Why are we actually skipping it? If I wanted to plot something from t0 to t1 with all steps in between I would probably also want to have the initial value in as well?

Reply:
The conventions on which matrices to skip are made such that they will work well with the broadcasted stochastic shift rule. However, I see that this behaviour is not super intuitive. The problem is that skipping matrix entries manually is not possible when setting up a broadcasted tape and sending it off for execution to a device.
I'm not sure what is best here.

Reply:
The idea is that the stochastic parameter-shift rule creates a tape with broadcasted times.
This makes the tape broadcasted. On hardware, it would be decomposed into multiple tapes again, on simulators the broadcasted matrix can be used if the device supports this. However, as the slicing you suggest would happen in the simulator device code, it would introduce rather non-local dependencies.

Reply:
I noticed just now that there will be problem with the `broadcasted tape` -> `multiple tapes` expansion.
Let me think some more about whether there is a neat solution.